### ISSUE TITLE
Remove Yandex trackers from the query string

### ISSUE THREAD

**Issue by fmarier** [brave/brave-browser#33216]
Yandex uses these query parameters in their analytics and ads products:
- `yadclid`
- `yadordid`
- `yclid`
- `ymclid`
- `ysclid`

# `yclid` and `ymclid`

Both `yclid` and `ymclid` are unique click identifiers according to this [post](https://osipenkov.ru/yclid-gclid-analytics/) ([translation](http://madik.ru/web-analitika/metki-gclid-yclid-ymclid-fbclid/)) and their documentation:
https://yandex.com/support/metrica/data/offline-params.html#offline-params__yclid
https://yandex.com/support/metrica/general/source-tags.html#sources-tags

It turns out we [already strip `yclid`](https://github.com/brave/brave-browser/issues/11578) because it's used by Yahoo! Japan, but we should strip `ymclid` too.

# `ysclid`

According to [this announcement](https://searchengines.guru/ru/news/2055667) ([translation](https://searchengines-guru.translate.goog/ru/news/2055667?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp)) and the [official](https://yandex.ru/support/webmaster/service/search-query-sup.html) [documentation](https://yandex-ru.translate.goog/support/webmaster/service/search-query-sup.html?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp), it's for tracking search terms when going from Yandex Search to a third-party website and so in theory it's not a user tracker per se.

I used Firefox to [search for "brave browser" on yandex.com](https://yandex.com/search/?text=brave+browser&lr=21353&search_source=yacom_desktop_common&src=suggest_B) and found this site with a `ysclid`:
- https://braveser.com/?ysclid=lmv7h6xnh4953068390

Then I repeated the same search in Private Browsing and got a different link
each time I moused over the link:
- https://braveser.com/?ysclid=lmv7jr4jhw515257938
- https://braveser.com/?ysclid=lmv7k17w6q715037679

Even the original window has a different ID now:
- https://braveser.com/?ysclid=lmv7kag9a7769772885

So it does not appear to be a stable identifier :confused: 

That said, we wouldn't otherwise leak out the search terms directly since we cap cross-origin referrers to the origin and so we should remove this too.

# `yadclid` and `yadordid`

While looking for examples of `yadclid` (which I found [here](https://community.cloudflare.com/t/please-add-a-new-yadclid-and-yclid-request-parameter-to-the-list-of-permissions/381411)), I also found a `yadordid` parameter:
- https://breezart.shop/solutions/breezart-1000-lux-ptc.php?yadclid=8040550&yadordid=17903971&yclid=13752475673004081151&action=SUBSCRIBE_PRODUCT&id=2423
- https://russian-seasons.ru/?utm_source=yandex&utm_medium=cpc&utm_campaign=jk_russkiye_sezoni&yadclid=97085288&yadordid=171563812&yclid=15851755138457796607
https://bolshayastrana.com/bajkal/znakomstvo-s-bajkalom-kruiz-23041?yadclid=55445739&yadordid=168122925
https://raylab.ru/catalog/svetofiltry/?utm_source=yandex&utm_medium=cpc&utm_campaign=yandex_cpc_raylab_search_all-goods_all-russia_age-all_device-all_none%7C73067181&utm_term=%D1%84%D0%B8%D0%BB%D1%8C%D1%82%D1%80+%D0%BF%D0%BE%D0%BB%D1%8F%D1%80%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D0%BE%D0%BD%D0%BD%D1%8B%D0%B9+raylab+cpl&yd_position=1&yd_source=none&yadclid=97189350&yadordid=173067181&ZGlyZWN0LnlhbmRleC5ydTs3MzA2NzE4MTsxMTk4Nzc1MjIwMDt5YW5kZXgucnU6cHJlbWl1bQ=&yclid=11775243539642843135&force=true&page=2

I've not been able to find any documentation about these online. Maybe they are unique identifiers? Maybe they are identifiers for a given creative, campaign or advertiser?

**Comment by fmarier** [comment]
All of these are removed by the "copy clean link" feature: https://github.com/brave/adblock-lists/pull/1320

**Comment by stephendonner** [comment]
## Verified `PASSED` using

```
Brave | 1.62.29 Chromium: 119.0.6045.105 (Official Build) nightly (x86_64)
-- | --
Revision | 722db0323f23c1aaee8a62326083bced61909279
OS | macOS Version 14.2 (Build 23C5030f)
```

### Steps:
1. installed `1.62.29`
2. launched Brave
3. loaded `https://brave.com/?foo=bar&irclickid=1234&ymclid=abcd&ysclid=efgh&fb_action_ids=5678&fb_comment_id=123456&clean=yes` with the `Developer Tools`' console open

### Confirmed entering and trying to load https://brave.com/?foo=bar&irclickid=1234&ymclid=abcd&ysclid=efgh&fb_action_ids=5678&fb_comment_id=123456&clean=yes resulted in only `https://brave.com/?foo=bar&clean=yes` being loaded

<img width="1897" alt="Screenshot 2023-11-06 at 3 22 59 PM" src="https://github.com/brave/brave-browser/assets/387249/8092c110-3baf-43f3-85db-9acd8cdfb204">

**Comment by kjozwiak** [comment]
Labelling as `QA/No` and removing `release-notes/include` as the above will be reverted via https://github.com/brave/brave-core/pull/21228 as per https://github.com/brave/brave-core/pull/21175 as per https://github.com/brave/brave-core/pull/21175#issue-2019827829.

**Comment by fmarier** [comment]
@stephendonner This will need a re-check once https://github.com/brave/brave-core/pull/21254 lands.

**Comment by kjozwiak** [comment]
Re-adding the `QA Pass` label as the above was already verified via `1.62.x`, We'll need to create a new issue that's associated with https://github.com/brave/brave-core/pull/21254 which will be in `1.63.x` so it can be re-verified on `1.63.x`. Technically nothing was reverted/changed via `1.62.x` so the above verification is still valid.

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on `Vivo X70 Pro version 12` running `Bravemonoarm64.apk_1.62.116`

- Verified the test plan from https://github.com/brave/brave-core/pull/21254
- When the URL `https://brave.com/?ymclid=1234&ysclid=abcde` is entered the two parameters `ysclid` and  `ymclid` were removed
- Confirmed that url bar just show the https://brave.com/
![image](https://github.com/brave/brave-browser/assets/38657976/4afd16da-b239-44f6-8d8b-7d0fbe3e5967)
